modules/puppet_connect/plans/test_input_data.pp

```diff
@@ -59,6 +60,16 @@
         fail_plan("Inventory contains target ${target} with unsupported transport, must be ssh or winrm")
       }
     }
+
+    $plugin = $target.plugin_hooks["puppet_library"]
+    $user_configured_plugin = $plugin != { "plugin"=> "puppet_agent", "stop_service"=> true }
+
+    if ($user_configured_plugin and $plugin["plugin"] != "task"){
+      fail_plan("Only task plugins are acceptable for puppet_library hook")
+    }
+    if ($target.plugin_hooks != $plugin_hook){
+      fail_plan("The puppet_library plugin config must be the same across all targets")
```

Comment:
We should probably include as much detail as possible and that is useful here, like in the Orch-side validation. At a minimum, I think the non-unique target's name would be good.